src/CLIHandler.ts

```diff
@@ -2,361 +2,362 @@
+					},
+				},
+			},
+		});
+
+		this._screen.append(this._details);
+
+		this._details.focus();
+	}
+
+	end(): void {
+		if (!this._screen) {
+			return;
+		}
+
+		this._screen.render();
+
+		const snapshot = this._screen.screenshot();
+		this._screen.destroy();
+		process.stdout.write(snapshot);
+	}
+
+	async progress(message: string, promise: PromiseOrCallback): Promise<any> {
+		let out;
+
+		if (this.interactive) {
+			const entry = new OverviewEntry({
+				top: this._entries++,
+			});
+			this._overview?.append(entry);
+			try {
+				entry.start(this._getPrefix() + message);
+
+				if (this.nestingLevel > 0) {
+					this.info(this._getPrefix() + message);
+				} else {
+					this.section(message);
+				}
+
+				this.nestingLevel++;
+
+				if (typeof promise === 'function') {
+					out = await promise();
+				} else {
+					out = await promise;
+				}
+				entry.end(true);
+				return out;
+			} catch (e) {
+				entry.end(false);
+				throw e;
+			} finally {
+				this.nestingLevel--;
+			}
+			return;
+		}
+
+		try {
+			this.info(message + ' - START');
+			this.nestingLevel++;
+
+			if (promise instanceof Function) {
+				out = await promise();
+			} else {
+				out = await promise;
+			}
+			this.nestingLevel--;
+			this.info(message + ' - OK');
+			return out;
+		} catch (e) {
+			this.nestingLevel--;
+			this.info(message + ' - FAILED');
+			throw e;
+		}
+	}
+
+	async check(
+		message: string,
+		ok: PromiseOrCallback | boolean
+	): Promise<boolean> {
+		const okType = typeof ok;
+
+		if (this.interactive) {
+			const entry = new OverviewEntry({
+				top: this._entries++,
+			});
+
+			entry.start(this._getPrefix() + message);
+
+			if (ok instanceof Function) {
+				ok = await ok();
+			}
+
+			if (ok instanceof Promise) {
+				ok = await ok;
+			}
+
+			entry.end(ok as boolean);
+			if (ok) {
+				this.info(this._getPrefix() + message + ' - OK');
+			} else {
+				this.warn(this._getPrefix() + message + ' - FAILED');
+			}
+
+			this._overview?.append(entry);
+			return ok;
+		}
+
+		if (ok instanceof Function) {
+			ok = await ok();
+		}
+
+		if (ok instanceof Promise) {
+			ok = await ok;
+		}
+
+		this._log('INFO', ['%s: %s', message, checkMark(ok as boolean)]);
+
+		return ok;
+	}
+
+	showValue(message: string, value: string): void {
+		if (this.interactive) {
+			const entry = new OverviewEntry({
+				top: this._entries++,
+			});
+
+			entry.start(this._getPrefix() + message + ': {bold}' + value + '{/}');
+			entry.end(true);
+
+			this._overview?.append(entry);
+			return;
+		}
+
+		this._log('INFO', ['%s: %s', message, value]);
+	}
+
+	info(message: string): void {
+		this._log('INFO', Array.from(arguments));
+	}
+
+	warn(message: string): void {
+		this._log('WARN', Array.from(arguments));
+	}
+
+	debug(message: string): void {
+		this._log('DEBUG', Array.from(arguments));
+	}
+
+	error(message: string): void {
+		this._log('ERROR', Array.from(arguments));
+		if (!this.interactive) {
+			process.exit(1);
+		}
```

Comment:
Question is if this should be `process.exit()` or `process.exitCode = 1` 🤔

Reply:
i.e. exit immediately or just set the exitCode for whenever the process terminates.